lputils.pas

```diff
@@ -380,30 +381,45 @@ procedure ExposeGlobals(Compiler: TLapeCompiler; HeaderOnly, DoOverride: Boolean
       'end;';
   end;
 
+var
+  Code: lpString;
 begin
   if (Compiler = nil) then
     Exit;
 
-  Compiler.addDelayedCode(
-    LapeDelayedFlags +
-    GetGlobalPtr()  + LineEnding +
-    GetGlobalName() + LineEnding +
-    GetGlobalVal()  + LineEnding +
-    VariantInvoke() + LineEnding +
-    ToString()
-  , '!addDelayedExpose');
+  Code := LapeDelayedFlags;
+
+  if Compiler.hasDefine('Lape_VariantInvoke') then
```

Comment:
I would prefer the addition of an `hasBaseDefine` function, rather than adjust `hasDefine` to also check for BaseDefines.